mlir/lib/Dialect/MIOpen/Transforms/ConvToGemmLowering.cpp

```diff
@@ -837,6 +842,32 @@ LogicalResult backwardData(Conv2DBwdDataOp op, PatternRewriter &b) {
       // Replace filter gemm with padded version
       gemmFilter = b.create<TransformOp>(loc, gemmFilter, padTransformAttr);
     }
+
+    // KPack for filter tensor.
+    gemmFilterKPack = gemmFilter;
+    if (KPack > 1) {
+      assert(!filterCheckPadGemmM && !filterCheckPadGemmK);
+      BottomUpCTBuilder sourceTransform = gemmFilterTransform;
+      TransformMapAttr sourceTransformAttr = gemmFilterTransformAttr;
+
+      Value source = gemmFilter;
+
+      BottomUpCTBuilder kpackGemmFilterTransform =
+          BottomUpCTBuilder::above(sourceTransform, sourceTransformAttr);
+      // Passthrough gemmG (dim 0) and gemmM (dim 2).
+      kpackGemmFilterTransform.passThrough(sourceTransform.endName(0));
+      kpackGemmFilterTransform.passThrough(sourceTransform.endName(2));
+      // Use Unmerge to split gemmK (dim 1) into gemmK and gemmKPack, place
+      // gemmKPack at dim 3.
+      int64_t gemmKLength = sourceTransform.endSize(1);
+      auto gemmKName = sourceTransform.endName(1);
+      kpackGemmFilterTransform.unmerge({gemmKName, "gemmKPack"}, {1, 3},
+                                       gemmKName, {gemmKLength / KPack, KPack});
+      TransformMapAttr kpackGemmFilterTransformAttr =
+          kpackGemmFilterTransform.get();
+      gemmFilterKPack =
+          b.create<TransformOp>(loc, source, kpackGemmFilterTransformAttr);
```

Comment:
I'd prefer this + line 1036 + below consolidated into a function that builds a K / Kpack transform.

https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/blob/395e80539e3128362b3a1ec6659aff81df597db2/mlir/lib/Dialect/MIOpen/Transforms/ConvToGemmLowering.cpp#L1335-L1349

Reply:
Will do.